### PR TITLE
GEOMESA-1346 Generate Ordered Bresenham Coordate List

### PR DESCRIPTION
**Pull request by aheyne**
genBresenhamCoordList now returns a list. GenerateLineCoordSet and generateLineCoordList
call genBresenhamCoordList and return their respective types.

Signed-off-by: Austin Heyne <aheyne@ccri.com>

### CODE REVIEW

File: geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/GridSnap.scala [hunk 1]
```diff
@@ -9,6 +9,7 @@
 
 package org.locationtech.geomesa.utils.geotools
 
+import com.sun.xml.internal.bind.v2.runtime.Coordinator
```
Comment (elahrvivaz) [inline geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/GridSnap.scala]: i think this is an extraneous import, no?

File: geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/GridSnap.scala [hunk 2]
```diff
@@ -97,13 +98,18 @@ class GridSnap(env: Envelope, xSize: Int, ySize: Int) {
...
 
   /** Generate a Set of Coordinates between two given Snap Coordinates which includes both start and end points*/
   def generateLineCoordSet(coordOne: Coordinate, coordTwo: Coordinate): Set[Coordinate] = {
-    genBresenhamCoordSet(i(coordOne.x), j(coordOne.y), i(coordTwo.x), j(coordTwo.y)) + coordOne + coordTwo
+    genBresenhamCoordList(i(coordOne.x), j(coordOne.y), i(coordTwo.x), j(coordTwo.y)).toSet + coordOne + coordTwo
+  }
+
+  /** Generate an ordered List of Coordinates between two given Snap Coordinates which includes both start and end points*/
+  def generateLineCoordList(coordOne: Coordinate, coordTwo: Coordinate): List[Coordinate] = {
+    coordOne +: genBresenhamCoordList(i(coordOne.x), j(coordOne.y), i(coordTwo.x), j(coordTwo.y)) :+ coordTwo
```
Comment (elahrvivaz) [inline geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/GridSnap.scala]: is there any possibility of coordOne or two being duplicated? In the 'generateLineCoordSet' method the set type will prevent that.

Comment (aheyne) [inline geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/GridSnap.scala]: It shouldn't but I've added .distinct in generateLineCoordList to be sure.

**Comment by jnh5y** [comment]
Manually merge-squashed.  Committed as c369e24b749662f1bda704225a1d2128d2e84387.

**Pull request closed without merging**